Engine/Animation/AnimController.cpp

```diff
@@ -132,19 +141,19 @@ void AnimController::FinishActiveState()
 	apply_transition = false;
 }
 
-void PlayingClip::Update()
+void PlayingClip::Update(float speed)
 {
 	if (!playing || !clip || clip->animation == nullptr)
 	{
 		return;
 	}
-	current_time = current_time + static_cast<int>(App->time->delta_time);
+	current_time = (current_time + (App->time->delta_time)* speed) ;
 	if (current_time >= clip->animation_time)
 	{
 		if (clip->loop)
 		{
 
-			current_time = current_time % clip->animation_time;
+			current_time = current_time / clip->animation_time;//before it was %
```

Comment:
Why are you taking now the quocient instead of the module?

Reply:
I converted current_time to float, thus I needed to make this change to division from quotient. This was because it is not allowed to use "%" with floats.

Reply:
Yeah, but the operation `%` is different from `/`. The former was done to adjust the current time while looping. Ej:
```
animation_time = 10;
current_time = 14;
// Because we are looping and exceding animation time, currentt ime needs to be between 0 and 9.
current_time = current_time % clip // current_time = 14 % 10 = 4
``` 
But, if we use `/` we have:
```
current_time = 14/10 = 1.4
```
Which is not correct. I understand that `float` has no % operation, but we should replicate this operators for floats. 

Let me know if I'm mistaken :D

Reply:
Yes I suppose you stand correct! I probably didn't think this through.